addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/discovery/AVMFritzDiscoveryService.java

```diff
@@ -79,12 +80,27 @@ public AVMFritzDiscoveryService(BoxHandler bridgeHandler) {
         this.activate(null);
     }
 
+    @Override
+    protected void activate(Map<String, Object> configProperties) {
```

Comment:
I'd assume that this method is never called as this class is not handled by DS.

Reply:
Maybe I misunderstood something, but didn't we register this discovery as a new service once a BoxHandler is created (see [HandlerFactory.java#L69](https://github.com/cweitkamp/openhab2-addons/blob/eb7518303774845dc669d1a974ccb7b0a78a6deb/addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/HandlerFactory.java#L69))?

Reply:
Yes, but you must not confuse the registration of a class instance as an OSGi service with declaring a service and handing over the lifecycle to DS.